### PR TITLE
chore: split DAL into two packages, with moduleconfig queries moved to common/configuration

### PR DESCRIPTION
**Pull request by deniseli**
Fixes https://github.com/TBD54566975/ftl/issues/1696

Approach:

* Duplicate `- engine` block of `sqlc.yaml` to support multiple `out` locations
* Refactor common error handling vars + funcs into separate `dalerrs` package. This now lives under the new top level `db` dir, where we can add additional common DB logic as we split the controller dal further. This change incurred a lot of NOOP lines changed. :/
* Meat of the change: split moduleconfig CRUD queries out of `backend/controller/sql` and into `common/configuration/sql`. Correspondingly, split the moduleconfig DAL functions out of `backend/controller/dal` and into `common/configuration/dal`.
* Questionable choice: Copied `backend/controller/sql/conn.go` to `common/configuration/sql/conn.go` with the unnecessary logic stripped out (`Tx`...). That means that what remains (`sql.DBI` and `sql.NewDB`), though a small amount of code, is duplicated with the original `conn.go`. I could refactor this to use a shared interface instead, but that could also make this messier for readability and take things in a worse direction overall, so pending better ideas, we can rethink this when splitting the controller DAL.

### CODE REVIEW

File: Justfile [hunk 1]
```diff
@@ -67,6 +67,7 @@ init-db:
 # Regenerate SQLC code (requires init-db to be run first)
 build-sqlc:
   @mk backend/controller/sql/{db.go,models.go,querier.go,queries.sql.go} : backend/controller/sql/queries.sql backend/controller/sql/schema sqlc.yaml -- "just init-db && sqlc generate"
+  @mk common/configuration/sql/{db.go,models.go,querier.go,queries.sql.go} : common/configuration/sql/queries.sql backend/controller/sql/schema sqlc.yaml -- "just init-db && sqlc generate"
```
Comment (alecthomas) [inline Justfile:70]: Combine these - mk can take any number of inputs and any number of outputs

File: sqlc.yaml [hunk 2]
```diff
@@ -137,6 +137,143 @@ sql:
...
+    database:
+      uri: postgres://localhost:15432/ftl?sslmode=disable&user=postgres&password=secret
+    gen:
+      go:
+        package: "sql"
+        sql_package: "pgx/v5"
+        out: "common/configuration/sql"
+        emit_interface: true
+        query_parameter_limit: 3
+        overrides:
```
Comment (alecthomas) [inline sqlc.yaml:152]: I mentioned this before, but can you deduplicate this with [YAML anchors](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/)?

**Review by alecthomas — APPROVED** [review]
This is awesome! I'm keen to start splitting it up further, it'll make the code a lot cleaner.

File: Justfile (hunk 1 above)
Comment (deniseli) [reply Justfile:70]: sweet! done

File: sqlc.yaml (hunk 2 above)
Comment (deniseli) [reply sqlc.yaml]: Oof thanks for the reminder! fixed

**Pull request merged**